api/main.py

```diff
@@ -165,19 +164,26 @@ async def get_node(node_id: str, kind: str = "node"):
         ) from error
 
 
-@app.get('/nodes', response_model=List[Node])
```

Comment:
Maybe the response type can still be a `List` without specifying the type of its elements?  Also it might still be valid if all the elements are sub-classes of `Node` but I'm not sure how FastAPI would deal with that.

Reply:
If we use `Node`, it will not return fields only specific to `Regression` model.
We also can not use only `List` as then it fails to serialize response:
```
kernelci-api |   File "/home/kernelci/.local/lib/python3.10/site-packages/fastapi/encoders.py", line 144, in jsonable_encoder
kernelci-api |     raise ValueError(errors)
kernelci-api | ValueError: [TypeError("'ObjectId' object is not iterable"), TypeError('vars() argument must have __dict__ attribute')]
```

Reply:
Well Python can handle that like any object-oriented language, if all the objects in the list are based on `Node`. For example:

```python
from typing import List


class A:
    name = 'something'


class B(A):
    name = "derived"
    foo = 'bar'


def main():
    l: List(A) = [A(), B()]
    print(l[0].name)
    print(l[1].name)
    print(l[1].foo)


if __name__ == '__main__':
    main()
```

But my question was whether Pydantic and FastAPI can handle that, by validating a `Regression` object and storing it in a list of `Node` object. I think it's worth double-checking.

Reply:
> > We also can not use only `List` as then it fails to serialize response
> 
> How did that work before when the returned type was a `List[Node]`?

I meant we can not use it without specifying any datatype i.e. `response_model=List[]`

Reply:
> Well Python can handle that like any object-oriented language, if all the objects in the list are based on `Node`. For example:
> 
> ```python
> from typing import List
> 
> 
> class A:
>     name = 'something'
> 
> 
> class B(A):
>     name = "derived"
>     foo = 'bar'
> 
> 
> def main():
>     l: List[A] = [A(), B()]
>     print(l[0].name)
>     print(l[1].name)
>     print(l[1].foo)
> 
> 
> if __name__ == '__main__':
>     main()
> ```
> 
> But my question was whether Pydantic and FastAPI can handle that, by validating a `Regression` object and storing it in a list of `Node` object. I think it's worth double-checking.

Yes, I did check. Here is the documentation URL from FastAPI: https://fastapi.tiangolo.com/tutorial/extra-models/#union-or-anyof 
The code with `response_model=List[Union[Regression,Node]]` is working as expected for Node and regression models.